mapo/agents/mapo/tests/test_losses.py

```diff
@@ -0,0 +1,208 @@
+# pylint: disable=missing-docstring, redefined-outer-name
+import pytest
+import tensorflow as tf
+from ray.rllib.models.catalog import ModelCatalog
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.evaluation.postprocessing import Postprocessing
+from ray.rllib.utils.tracking_dict import UsageTrackingDict
+
+
+import mapo.agents.mapo.losses as losses
+from mapo.agents.mapo import MAPOModel
+
+
+@pytest.fixture
+def batch_tensors(spaces):
+    obs_space, action_space = spaces
+    obs = tf.compat.v1.placeholder(
+        tf.float32, shape=[None] + list(obs_space.shape), name="observation"
+    )
+    actions = ModelCatalog.get_action_placeholder(action_space)
+    rewards = tf.compat.v1.placeholder(tf.float32, [None], name="reward")
+    dones = tf.compat.v1.placeholder(tf.bool, [None], name="done")
+    next_obs = tf.compat.v1.placeholder(
+        tf.float32, shape=[None] + list(obs_space.shape), name="next_observation"
+    )
+    returns = tf.compat.v1.placeholder(tf.float32, [None], name="return")
+    return UsageTrackingDict(
+        {
+            SampleBatch.CUR_OBS: obs,
+            SampleBatch.ACTIONS: actions,
+            SampleBatch.REWARDS: rewards,
+            SampleBatch.DONES: dones,
+            SampleBatch.NEXT_OBS: next_obs,
+            Postprocessing.ADVANTAGES: returns,
+        }
+    )
+
+
+@pytest.fixture
+def model_config():
+    return {
+        "custom_options": {
+            "actor": {"activation": "relu", "layers": [32, 32]},
+            "critic": {"activation": "relu", "layers": [32, 32]},
+            "dynamics": {"activation": "relu", "layers": [32, 32]},
+        }
+    }
+
+
+@pytest.fixture(params=[True, False])
+def target_networks(request):
+    return request.param
+
+
+@pytest.fixture(params=[True, False])
+def smooth_target_policy(request):
+    return request.param
+
+
+@pytest.fixture(params=[True, False])
+def twin_q(request):
+    return request.param
+
+
+@pytest.fixture
+def kernel_fn():
+    return lambda x, y: tf.norm(x - y, axis=0)
+
+
+# skip true dynamics test until env dynamics log_prob is done
+@pytest.fixture(params=[False, pytest.param(True, marks=pytest.mark.skip)])
```

Comment:
Will code `env.compute_log_prob` to unskip that just after this review...

Reply:
Done! If you accept the feature/env-log-prob PR I've just opened you might be able to run these tests...